### ISSUE TITLE
[BUG/ISSUE] Full-chem simulation fails for January start dates in 12.7.0 if SfcVMR true

### ISSUE THREAD

**Issue by jaegle** [geoschem/geos-chem#233]
## Describe the bug
I downloaded GEOS-Chem v12.7.1 and was able to run a standard simulation from 20160701 until 20170101. In order to spin up the simulation for another year, I copied the file restart file I obtained (GEOSChem. Restart.20170101_0000z.nc4) to a new restart file called GEOSChem.Restart.20160101_0000z.nc4. I then changed the date in input.geos to 20160101 (see below):
```
%%% SIMULATION MENU %%% :
Start YYYYMMDD, hhmmss  : 20160101 000000
```
When I start the simulation again with this new restart file, the run stops while trying to read CMIP6_GHG_surface_VMR_2014.nc

output from HEMCO.log:
```
--> LOCATION: HCOIO_READ_STD (hcoio_read_std_mod.F90)
HEMCO WARNING: Data is treated as unitless, but file attribute suggests it is not: mol mol-1 dry. File: ./GEOSChem.Restart.20160101_0000z.nc4
--> LOCATION: HCOIO_READ_STD (hcoio_read_std_mod.F90)
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
HEMCO ERROR: Invalid time index in /home/disk/oreo/data/ctm/HEMCO/CMIP6/v2019-12/CMIP6_GHG_surface_VMR_2014.nc
```

Do I need to change the date/time in the restart file? If so how do I do it? 

## Required information
Please include the following:
 - 12.7.1

## Input and log files to attach
[GC.log](https://github.com/geoschem/geos-chem/files/4273567/GC.log)
[HEMCO.log](https://github.com/geoschem/geos-chem/files/4273569/HEMCO.log)
[HEMCO.log](https://github.com/geoschem/geos-chem/files/4273570/HEMCO.log)

**Comment by msulprizio** [comment]
Renaming the GEOS-Chem restart file as you described should have worked. As you point out, the run crashes when trying to read the CMIP6_GHG_surface_VMR_2014.nc file. The CMIP4 CH4 data should only be read if the NOAA GMD data are not available (i.e. if you are running prior to 1979). Can you check the top of your HEMCO_Config.rc and make sure you have:

```
    --> GMD_SFC_CH4            :       true
    --> CMIP6_SFC_CH4          :       false
```

I'd also recommend setting `Verbose` and `Warnings` to 3 at the top of HEMCO_Config.rc to get additional information in your HEMCO.log file. If you're still having issues, after checking those settings please attach the HEMCO.log file with verbose and warnings output and your HEMCO_Config.rc file.

**Comment by jaegle** [comment]
Hello Melissa,

Thanks for your answer. I checked HEMCO_Config.rc and it looks like it has the correct flags for GMD_SFC_CH4/CMIP6_SFC_CH4.

I ran GEOS-Chem again with the verbose and warnings flags. Here are the HEMCO config and log files I obtained.

Lyatt
[HEMCO.log](https://github.com/geoschem/geos-chem/files/4277550/HEMCO.log)
[HEMCO_Config.rc.txt](https://github.com/geoschem/geos-chem/files/4277552/HEMCO_Config.rc.txt)

**Comment by yantosca** [comment]
I see that you have in HEMCO_Config.rc:
```
    --> GMD_SFC_CH4            :       true
    --> CMIP6_SFC_CH4          :       false
```
which definitely match the brackets further down in the HEMCO_Config.rc file:
```
```
(((GMD_SFC_CH4
* NOAA_GMD_CH4 $ROOT/NOAA_GMD/v2018-01/monthly.gridded.surface.methane.1979-2020.1x1.nc SFC_CH4 1979-2020/1-12/1/0 C xy ppbv * - 1 1
)))GMD_SFC_CH4
. . .
(((CMIP6_SFC_CH4
* CMIP6_Sfc_CH4  $ROOT/CMIP6/v2019-12/CMIP6_GHG_surface_VMR_$YYYY.nc CH4 1750-1978/1-12/1/0 C xy ppbv * - 1 1
)))CMIP6_SFC_CH4
```
Nevertheless, HEMCO still is trying to read the CMIP6_SFC_CH4 data even though it has been switched off.

One thing you can do is to try to comment that out in the HEMCO_Config.rc file and to see if the error still persists:

```
#  --> CMIP6_SFC_CH4          :       false
...
#(((CMIP6_SFC_CH4
#* CMIP6_Sfc_CH4  $ROOT/CMIP6/v2019-12/CMIP6_GHG_surface_VMR_$YYYY.nc CH4 1750-1978/1-12/1/0 C xy ppbv * - 1 1
#)))CMIP6_SFC_CH4
```
I would also check to make sure that your HEMCO_Config.rc isn't getting overwritten or that it GC isn't trying to read another HEMCO_Config.rc from a different folder.  It could be something like that.  Keep us posted.

**Comment by lizziel** [comment]
I think the issue is reading CH3Cl which is read if SfcVMR is turned on, which it is in your case. SfcVMR_CH3Cl is the first HEMCO container after NOAA_GMD_CH4 based on the HEMCO log file with verbose on. Your GC.log file has this to confirm it:
```
HEMCO: Opening /home/disk/oreo/data/ctm/HEMCO/CMIP6/v2019-12/CMIP6_GHG_surface_VMR_2014.nc
 Error in HCOIO_DATAREAD called from HEMCO ReadList_Fill: SfcVMR_CH3Cl
```

The times in the file are these (units are days):
` time = 15, 46, 74, 105, 135, 166, 196, 227, 258, 288, 319, 349 ; `

I wonder if there is an issue that the times in the file are the middle of the month. Do you get the same issue if you try to start the simulation on January 15th?

**Comment by jaegle** [comment]
The issue does seem to occur when reading the SfcVMR_CH3Cl files. When I comment these lines out in HEMCO_Config:
#==============================================================================
# --- CMIP6 files ---
#
# Use core CMIP6 observationally constrained data from Meinshausen et al. (2017) GMD
#==============================================================================
* SfcVMR_CH3Cl  $ROOT/CMIP6/v2019-12/CMIP6_GHG_surface_VMR_$YYYY.nc CH3Cl  1750-2014/1-12/1/0 C xy ppbv * 801 1 1
* SfcVMR_CH2Cl2 $ROOT/CMIP6/v2019-12/CMIP6_GHG_surface_VMR_$YYYY.nc CH2Cl2 1750-2014/1-12/1/0 C xy ppbv * 801 1 1
* SfcVMR_CHCl3  $ROOT/CMIP6/v2019-12/CMIP6_GHG_surface_VMR_$YYYY.nc CHCl3  1750-2014/1-12/1/0 C xy ppbv * 801 1 1
* SfcVMR_CH3Br  $ROOT/CMIP6/v2019-12/CMIP6_GHG_surface_VMR_$YYYY.nc CH3Br  1750-2014/1-12/1/0 C xy ppbv * 801 1 1

the run proceeds (but I assume no surface values of CH3Cl are read in). 

If I start the run on a day other than Jan 1 I have no issues (I tried Feb 1, Feb 15, and Aug 1 2016 - which all worked). So maybe there is indeed an issue with the times in the CMIP6_GHG surface vmr files.

**Comment by lizziel** [comment]
Thanks for getting back so quickly. I wonder if this is an edge case for January, and the same issue may occur for the other edge case in late December.

**Comment by lizziel** [comment]
I am able to reproduce this issue. I'll post here when we have a fix.

**Comment by lizziel** [comment]
I renamed this issue to capture the full extent of the problem. The CMIP6 files have mid-month timestamps but HEMCO_Config.rc specifies day 1 of each month as the time slice to be used. Switching to say, day 15, will not work because the day per month is not fixed. In addition, the files are annual so data prior to mid-month will not be available.

We are in touch with the developer to find out the intended use of the timestamp. If it represents simply the month the value should be used for we can adjust the time values to correspond to the first of each month. If it represents instead the first day of the month for each value then we will need to either include the previous December mid-month value per annual file or bundle all years together into a single file.

**Comment by lizziel** [comment]
From Tomas Sherwin:
> The files contain 12 data entries, one for each month. the times are set as the middle of the month, but the data is to be used for the entire of a given month. Does that clear this up?

We can therefore simply change the timestamps in the file to be the first of each month to fix this issue. However, this does mean that the files are used incorrectly in 12.7 since the first half of each month used the previous month’s data. Since the fix will change the benchmark we will put it into the next benchmarked version which is 12.8.

**Comment by lizziel** [comment]
CMIP6 data with time shifted from middle of month to first day of every month will be available on the [Harvard ftp site](http://ftp.as.harvard.edu/gcgrid/data/ExtData/HEMCO/CMIP6/v2020-03/) later today. You can change the path of the CMIP6 files in HEMCO_Config.rc to use the new data directory (v2020-03 rather than v2019-12) as a quick fix in 12.7.
